binary_transparency/firmware/cmd/flash_tool/flash_tool.go

```diff
@@ -126,26 +122,36 @@ func readUpdateFileFromFlags() (api.UpdatePackage, error) {
 func checkSignature(up api.UpdatePackage) error {
 	stmt := api.FirmwareStatement{}
 	if err := json.NewDecoder(bytes.NewReader(up.ProofBundle.ManifestStatement)).Decode(&stmt); err != nil {
-		return fmt.Errorf("failed to decode firmware statement: %q", err)
+		return fmt.Errorf("failed to decode firmware statement: %w", err)
 	}
 
 	//Verify the signature:
 	if err := crypto.VerifySignature(stmt.Metadata, stmt.Signature); err != nil {
-		return fmt.Errorf("firmware signature verification failed reason %q", err)
+		return fmt.Errorf("firmware signature verification failed reason %w", err)
 	}
 	return nil
 }
 
 // verifyUpdate checks that an update package is self-consistent.
-func verifyUpdate(up api.UpdatePackage) error {
-	fwHash := sha512.Sum512(up.FirmwareImage)
-	if err := verify.BundleForUpdate(up.ProofBundle, fwHash[:]); err != nil {
-		return fmt.Errorf("failed to verify proof bundle: %q", err)
+func verifyUpdate(c *client.ReadonlyClient, up api.UpdatePackage, dev devices.Device) error {
+	// Get the consistency proof for the bundle
+	dc, err := dev.DeviceCheckpoint()
+	if err != nil {
+		return fmt.Errorf("failed to fetch the device checkpoint: %w", err)
 	}
-	return nil
-}
 
-func checkConsistency(c *client.ReadonlyClient, up api.UpdatePackage, dev devices.Device) error {
-	// TODO(al): implement this
+	var proof [][]byte
```

Comment:
Might be worth calling this `cp` for clarity (can change the cp below to `r` or similar).